src/quotamodel.c

```diff
@@ -1503,6 +1504,12 @@ do_load_quotas(void)
 			targetOid = primaryOid;
 		}
 
+		if (!SearchSysCacheExists1(quota_key_caches[quotaType][0], ObjectIdGetDatum(targetOid)))
```

Comment:
Maybe check that quotaType is in the correct range before using it in this way?

Reply:
I've added the check. Superuser can add entries with invalid values in diskquota.quota_config.quotaType by mistake